mne/io/fiff/tests/test_raw.py

```diff
@@ -888,9 +888,16 @@ def test_hilbert():
     picks = picks_meg[:4]
 
     raw2 = raw.copy()
+    raw3 = raw.copy()
+    raw4 = raw.copy()
     raw.apply_hilbert(picks)
     raw2.apply_hilbert(picks, envelope=True, n_jobs=2)
 
+    # Test custom N
+    raw3.apply_hilbert(picks, N=raw3.n_times + 100)
+    assert_equal(raw3._data.shape, raw._data.shape)
```

Comment:
It seems like maybe this is not as big of an issue for bandpass filtered signals? Maybe the differences between them are within the tolerance that matters? Not sure how to check for this... e.g.:

```python
f, (ax1, ax2) = plt.subplots(2, 1, figsize=(15, 10))
raw.filter(20, 40)
out1 = sg.hilbert(raw._data[0, :1000])
out2 = sg.hilbert(raw._data[0, :1000], N=1500)[:1000]
out3 = sg.hilbert(raw._data[0, :1000], N=5000)[:1000]

for dat in [out1, out2, out3]:
    print(dat[:5])
    print(np.abs(dat[:5]))
    print('\n')
    ax1.plot(np.abs(dat[:200]))
    ax1.plot(dat[:200])
    ax2.plot(np.abs(dat[-200:]))
    ax2.plot(dat[-200:])
    
```


![image](https://cloud.githubusercontent.com/assets/1839645/9429661/d3d2de68-498c-11e5-8a1e-f5fc8f268b79.png)